IPython/core/magics/basic.py

```diff
@@ -176,10 +176,18 @@ def alias_magic(self, line=''):
                 magic_escapes['cell'], name,
                 magic_escapes['cell'], target, params_str))
 
+    @magic_arguments.magic_arguments()
+    @magic_arguments.argument(
+        "-j", "--json", action="store_true", help="Output as JSON"
+    )
     @line_magic
     def lsmagic(self, parameter_s=''):
         """List currently available magic functions."""
-        return MagicsDisplay(self.shell.magics_manager, ignore=[])
+        args = magic_arguments.parse_argstring(self.lsmagic, parameter_s)
+        md = MagicsDisplay(self.shell.magics_manager, ignore=[])
+        if args.json:
+            return md._repr_json_()
+        return md
```

Comment:
Since this is a simple either/or, I would suggest the slightly more immediately explicit

```python
if args.json:
    return md._repr_json_()
else:
    return md
```

But otherwise looks good to me.

Thanks!!

Reply:
@fperez 
Thanks, for your review.
I modified at https://github.com/ipython/ipython/pull/14331/commits/36a3759d56f85d616ed7a3204b5499213e894bf9